src/App.tsx

```diff
@@ -1,24 +1,230 @@
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorOfUpdate, setErrorOfUpdate] = useState('');
+  const [isTodosLoading, setIsTodosLoading] = useState(true);
+  const [filterOption, setFilterOption] = useState(MainFilter.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [textOfInput, setTextOfInput] = useState('');
+  const [loadingIds, setLoadingIds] = useState<number[]>([]);
+
+  const numberOfActiveTodos = todos
+    .filter(({ completed }) => !completed).length;
+
+  const numberOfCompletedTodos = todos
+    .filter(({ completed }) => completed).length;
+
+  const isInputDisabled = !!tempTodo && !!loadingIds.length;
+
+  const getTodosFromServer = async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch {
+      setErrorOfUpdate('Something was wrong, during the loading...');
+    } finally {
+      setIsTodosLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    setErrorOfUpdate('');
+    getTodosFromServer();
+    setTimeout(() => setErrorOfUpdate(''), 3000);
+  }, []);
+
+  const getFilteredTodos = (filter: MainFilter, someTodos: Todo[]) => {
+    switch (filter) {
+      case MainFilter.Active:
+        return someTodos.filter((todo) => !todo.completed);
+      case MainFilter.Completed:
+        return someTodos.filter((todo) => todo.completed);
+      default:
+        return someTodos;
+    }
+  };
+
+  const addNewTodo = async (title: string) => {
+    const newTodo = {
+      title,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    if (title.trim() === '') {
+      setErrorOfUpdate('Title can\'t be empty');
+      setTextOfInput('');
+      setTimeout(() => {
+        setErrorOfUpdate('');
+      }, 3000);
+
+      return;
+    }
+
+    setTempTodo({ ...newTodo, id: 0 });
+
+    try {
+      const newTempTodo = await addTodo(newTodo);
+
+      setTodos(prevTodos => ([...prevTodos, newTempTodo]));
+    } catch {
+      setErrorOfUpdate('Unable to add a todo');
+    } finally {
+      setTempTodo(null);
+      setTextOfInput('');
+    }
+  };
+
+  const deleteTodoItem = async (todoId: number) => {
+    try {
+      setLoadingIds(state => [...state, todoId]);
+      await deleteTodo(todoId);
+      setTodos(todos.filter(todo => todo.id !== todoId));
+    } catch {
+      setErrorOfUpdate('Unable to delete a todo');
+      setTimeout(() => {
+        setErrorOfUpdate('');
+      }, 3000);
+    } finally {
+      setLoadingIds([]);
+    }
+  };
+
+  const clearCompletedTodos = () => {
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    setLoadingIds(completedTodos.map(todo => todo.id));
+
+    completedTodos.forEach(todo => {
+      deleteTodo(todo.id)
+        .then(() => {
+          setTodos(todos.filter(todoItem => !todoItem.completed));
+        })
+        .catch(() => {
+          setErrorOfUpdate('Unable to delete todos');
+        });
+    });
```

Comment:
Consider using Promise.all to handle multiple deleteTodo calls instead of chaining them with forEach. This will improve performance by executing the Promises concurrently.